homeassistant/components/knx/fan.py

```diff
@@ -32,18 +43,21 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 class KNXFan(KnxEntity, FanEntity):
     """Representation of a KNX fan."""
 
-    def __init__(self, device: XknxFan):
+    def __init__(self, device: XknxFan) -> None:
         """Initialize of KNX fan."""
+        self._device: XknxFan
         super().__init__(device)
 
+        self._step_range: tuple[int, int] | None
         if self._device.mode == FanSpeedMode.STEP:
+            assert device.max_step is not None
```

Comment:
Asserts are indicators of:

- Upstream typing incorrect.
- Incorrect handling right here.

Either way, asserting isn't the solution, as it is removed when Python is creating optimized byte code.

Reply:
The FanSpeedMode Enum is just a marker if `device.max_step` is set or None. We can just directly use `max_step` and instead of FanSpeedMode and unwrap it this way.